src/services/cache.ts

```diff
@@ -0,0 +1,14 @@
+import { Service } from "../common";
+
+export class CacheService extends Service {
+  async call(path: string): Promise<unknown> {
+    if (!this.ctx.cache.useCache || !this.ctx.cache.url) {
+      return undefined;
+    }
+    const cachedCall = await fetch(`${this.ctx.cache.url}/v1/chains/${this.chainId}/${path}`);
+    if (cachedCall.status !== 200) {
+      return undefined;
```

Comment:
Should we print a log if the cache is failing?

Reply:
good point, have added a `console.warn` for the case where the call to the cache fails